libraries/AP_HAL_Linux/RCOutput_PCA9685.cpp

```diff
@@ -54,20 +55,19 @@ using namespace Linux;
 
 #define PWM_CHAN_COUNT 16
 
-static const AP_HAL::HAL& hal = AP_HAL::get_HAL();
+static const AP_HAL::HAL &hal = AP_HAL::get_HAL();
 
-RCOutput_PCA9685::RCOutput_PCA9685(uint8_t addr,
-                                             bool external_clock,
-                                             uint8_t channel_offset,
-                                             int16_t oe_pin_number) :
-    _i2c_sem(NULL),
+RCOutput_PCA9685::RCOutput_PCA9685(AP_HAL::OwnPtr<AP_HAL::I2CDevice> dev,
+                                   bool external_clock,
+                                   uint8_t channel_offset,
+                                   int16_t oe_pin_number) :
     _enable_pin(NULL),
     _frequency(50),
     _pulses_buffer(new uint16_t[PWM_CHAN_COUNT - channel_offset]),
-    _addr(addr),
```

Comment:
I would add `_dev(std::move(dev)` in this line since it follows the parameters order.